sotodlib/wiregrid/calibrate.py

```diff
@@ -0,0 +1,404 @@
+    elif len(np.shape(x)) == 3:
+        real = x[0]
+        imag = x[1]
+        N, M = np.shape(real)
+        A = params[0]
+        B = params[1]
+        C = params[2]
+        return (real2 - np.diag(A))**2 + (imag2 - np.diag(B))**2 - np.reshape(np.tile(C, M), (M, N)).T
+
+def _comp_plane_fit(obs_data, std_data, fitfunc, param0):
+    """
+    Fit the data with a circle in a complex plane
+
+    Parameters:
+    - obs_data : ndarray of Q and U signal with 16 steps in nominal, np.array([Q(16), U(16)])
+    - std_data : ndarray of the standard deviations of Q and U, np.array([Qerr(16), Uerr(16)])
+    - fitfunc : a function used for this fitting, basically assumed the function of _circle_resid
+    - param0 : the initial paramer set
+
+    Return:
+    - fit result determined by the implicit fitting of scipy.odr
+    """
+    if len(np.shape(obs_data)) == 2:
+        mdr = odr.Model(fitfunc, implicit=True)
+        mydata = odr.RealData(obs_data, y=1, sx=std_data)
+        myodr = odr.ODR(mydata, mdr, beta0=param0)
+        myoutput = myodr.run()
+        return myoutput
+    elif len(np.shape(obs_data)) == 3:
+        alloutput = []
+        for i in range(np.shape(obs_data)[1]):
+            mdr = odr.Model(fitfunc, implicit=True)
+            mydata = odr.RealData(obs_data[:, i, :], y=1, sx=std_data[:, i, :])
+            myodr = odr.ODR(mydata, mdr, beta0=param0[:, i])
+            myoutput = myodr.run()
+            alloutput.append(myoutput)
+        return alloutput
+    else:
+        print("This input vector is not valid shape.")
+        return False
+
+def _linear1d(x, a, b):
+    """
+    one dimention linear funciton mod by 360 [degee]
+    """
+    return (a*x + b)%360
+
+def wg_get_cfitres(aman):
+    """
+    Get the results by the circle fitting about the responce against the wires in Q+iU plane
+
+    Parameters:
+    - aman : AxisManager
+
+    Return:
+    - aman : AxisManager
+        This includes fit results for all the inpput detectors
+        cfitval : Estimated parameter values by fitting
+        cfitcov : covariance matrix of the estimated parameters
+        cfitresvar : Residual variance
+    """
+    obs_data = []
+    obs_std = []
+    fit_results = []
+    cfitval = []
+    cfitcov = []
+    cfitresvar = []
+    for _i in range(np.shape(aman.det_info)[0]):
+        obs_data.append(np.array([aman.wg.Q[:, _i], aman.wg.U[:, _i]]))
+        obs_std = np.array([aman.wg.Qerr[:, _i], aman.wg.Uerr[:, _i]])
+        params_init = _get_initial_param_circle(obs_data[-1])
+        fit_results.append(
+            _comp_plane_fit(obs_data[-1], obs_std, _circle_resid, params_init))
+        cfitval.append(
+            np.array([fit_results[-1].beta[0], fit_results[-1].beta[1], np.sqrt(fit_results[-1].beta[2])]))
+        cfitcov.append(fit_results[-1].cov_beta)
+        cfitresvar.append(fit_results[-1].res_var)
+        pass
+    aman.wg.wrap('cfitval', np.array(cfitval), [(0, 'dets')])
```

Comment:
You need to make a new label_axis in your axis manager with the name of the fit parameters (i.e. `A, B, C`) and when you wrap the cfitval, cfitcov, and cfitresvar you need to wrap them to those axes appropriately.